src/kOS/AddOns/RemoteTech/RemoteTechAntennaModuleFields.cs

```diff
@@ -118,28 +117,31 @@ protected new Structure GetKSPFieldValue(StringValue suffixName)
 
         private Guid GetTargetGuid(Structure target)
         {
-            var str = target as StringValue;
-            if (str != null)
+            var api = RemoteTechHook.Instance;
+            if (target is StringValue)
             {
-                string targetString = str;
+                string targetString = target.ToString();
                 if (targetString.Equals(NoTargetString, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    return new Guid(NoTargetGuid);
+                    return api.GetNoTargetGuid();
                 }
                 else if (targetString.Equals(ActiveVesselString, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    return new Guid(ActiveVesselGuid);
-                }
-                else if (targetString.Equals(MissionControlString, StringComparison.InvariantCultureIgnoreCase))
-                {
-                    return new Guid(MissionControlGuid);
+                    return api.GetActiveVesselGuid();
                 }
                 else
                 {
-                    var body = FlightGlobals.Bodies.FirstOrDefault(b => b.bodyName.Equals(targetString, StringComparison.InvariantCultureIgnoreCase));
+                    IEnumerable<string> groundStations = api.GetGroundStations();
+                    foreach (var groundStation in groundStations) {
+                        if (targetString.Equals(groundStation)) {
```

Comment:
I've made them case-sensitive on purpose. Those names are case-sensitive in RemoteTech. We want to be able to differentiate between `Site` and `site` in kOS.